dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hdfs/ScanContext.java

```diff
@@ -79,23 +100,38 @@ void endWalkDir(FileStatus dir, long nFiles, long nDirs, long accumFileSize) thr
       this.numDirs += nDirs;
       this.accumulatedFileSize += accumFileSize;
 
+      String absolutePath = dir.getPath().toUri().getPath();
+      HdfsFileStatus hdfsFileStatus = dfsClient.getFileInfo(absolutePath);
+      byte byteStoragePolicy = hdfsFileStatus.getStoragePolicy();
+      StoragePolicy storagePolicy = StoragePolicy.valueOf(byteStoragePolicy);
+
       csvPrinter.printRecord(
-          dir.getPath().toUri().getPath(),
+          absolutePath,
           dir.getOwner(),
           dir.getGroup(),
-          dir.getPermission().toString());
+          dir.getPermission(),
+          DATE_FORMAT.format(Instant.ofEpochMilli(dir.getModificationTime())),
+          nFiles,
+          nDirs,
+          storagePolicy != null ? storagePolicy.toString() : String.valueOf(byteStoragePolicy));
     }
   }
 
   void walkFile(FileStatus file) {}
 
   /** This method is used to produce meaningful metrics for log/debug purposes. */
   String getFormattedStats() {
+
     final Duration timeSinceScanBegin = Duration.between(instantScanBegin, Instant.now());
-    final Duration avgTimeSpentInListStatusPerFile =
-        numFilesByListStatus > 0
-            ? timeSpentInListStatus.dividedBy(numFilesByListStatus)
-            : Duration.ZERO;
+    Duration avgTimeSpentInListStatusPerFile;
+    long secondsSpentInListStatus;
+    synchronized (LOCK1) {
```

Comment:
Consider renaming to `LOCK` and adding `@GuardedBy` annotations.

Reply:
Is it `com.google.errorprone.annotations.concurrent.GuardedBy` ?

Reply:
I think it's `javax.annotation.concurrent.GuardedBy;` like in ExecutorManager